autoload/go/test.vim

```diff
@@ -122,13 +122,53 @@ function! go#test#Func(bang, ...) abort
   call call('go#test#Test', args)
 endfunction
 
+" Testcase runs a single subtest of a function for a subtest name on the
+" current cursor line.
+" Arguments are passed to the `go test` command.
+function! go#test#Case(bang, ...) abort
+  let l:testParent = go#util#TestName()
+  if l:testParent == ""
+    call go#util#EchoWarning("[test] no test found immediate to cursor")
+    return
+  endif
+
+  " Check for a struct initialization on the line, and take its value
+  " e.g. `name: \"test name\",` would extract 'testName'
+  let l:subTestName = go#test#subtestUnderCursor(getline("."))
+  if l:subTestName == ""
+    call go#util#EchoError('[test] no name for test case found on cursor line')
+    return
+  endif
+
+  " the pattern after l:testParent will allow testcases that are nested
+  " arbitrarily deep to be selected to run
+  let l:testParentPattern = l:testParent . "/\([^/]\+/\)*"
+
+  let l:args = [a:bang, 0, "-run", l:testParentPattern . "/" . l:subTestName . "$"]
```

Comment:
The argument to `-run` will have two slashes in it now. I think what's needed is either pass `l:testParentPattern . l:subTestName` as the argument to `-run` or to change the value assigned to `l:testParentPattern` two lines up to `l:testParent . '(/[^/]+)*`